File: vscode/src/completions/tree-sitter/query-tests/test-data/singleline-triggers.snap.ts
// 
// | - query start position in the source file.
// █ – query start position in the annotated file.
// ^ – characters matching the last query result.
//
// ------------------------------------

// Matches empty interface

  interface Interface {}
//^^^^^^^^^^^^^^^^^^^^^^ trigger[1]
//                    █

// Nodes types:
// trigger[1]: interface_declaration

// ------------------------------------
// Matches interface with missing closing brace

  interface IncompleteInterface {
//^ start trigger[1]
//                              █
  console.log('incomplete')
//                        ^ end trigger[1]

// Nodes types:
// trigger[1]: interface_declaration

// ------------------------------------
// Matches empty type

  type Type = {}
//^^^^^^^^^^^^^^ trigger[1]
//            █

// Nodes types:
// trigger[1]: type_alias_declaration

// ------------------------------------
// TODO: this case is not covered because the AST produced starts with the ERROR node.
// Does not match anything as it is invalid syntax

type IncompleteType = {
//                    |
console.log('incomplete')


// ------------------------------------
// Does not match non-empty interface

interface Interface {
    //              |
    name: string
}

File: vscode/src/completions/tree-sitter/query-tests/test-data/singleline-triggers.snap.go
// 
// | - query start position in the source file.
// █ – query start position in the annotated file.
// ^ – characters matching the last query result.
//
// ------------------------------------

  package main

// Matches empty struct

  type Creature struct {
//              ^ start trigger[1]
//                     █
  }
//^ end trigger[1]

// Nodes types:
// trigger[1]: struct_type

// ------------------------------------
// Does not match non-empty struct

type Animal struct {
	//             |
	Name string
	Type string
}

// ------------------------------------
// Matches empty struct

  var person = struct {
//             ^ start trigger[1]
//                    █
  }{}
//^ end trigger[1]

// Nodes types:
// trigger[1]: struct_type

// ------------------------------------
// Does not match non-empty struct

var human = struct {
	//             |
	FirstName string
	LastName  string
}{
	FirstName: "John",
	LastName:  "Doe",
}

// ------------------------------------
// Does not match function declaration

func greet() {
	//       |
}

// ------------------------------------
// Matches empty nested interface

  func wrap() {
      type Writer interface {
//                ^ start trigger[1]
//                          █
      }
//    ^ end trigger[1]
  }

// Nodes types:
// trigger[1]: interface_type

